src/test/ui/impl-trait/equality.stderr

```diff
@@ -1,6 +1,9 @@
 error[E0308]: mismatched types
   --> $DIR/equality.rs:25:5
    |
+21 | fn two(x: bool) -> impl Foo {
+   |                    -------- expected `i32` because of this return type
```

Comment:
This doesn't seem right. It's the `return 1_i32;` which is causing this error and ideally that's what would be pointed to.